mlrun/serving/states.py

```diff
@@ -955,9 +957,34 @@ def plot(self, filename=None, format=None, source=None, **kw):
 
 
 class Model(storey.ParallelExecutionRunnable):
+    def __init__(
+        self,
+        name: str,
+        raise_exception: bool = True,
+        artifact: Union[
+            ModelArtifact, Any, str
+        ] = None,  # TODO: Replace Any with LLMPromptArtifact when ready
+        **kwargs,
+    ):
+        super().__init__(name=name, raise_exception=raise_exception, **kwargs)
+        # TODO: Add LLMPromptArtifact when ready
+        if artifact is not None and not isinstance(artifact, (ModelArtifact, str)):
+            raise ValueError(
+                "Model support only ModelArtifact, LLMPromptArtifact or artifact uri(str) as model artifact."
+            )
+        self.artifact = artifact
+        self._kwargs = kwargs
+        self.model = None
+
     def load(self) -> None:
         """Override to load model if needed."""
-        pass
+        if isinstance(self.artifact, str):
+            if mlrun.datastore.is_store_uri(self.artifact):
+                self.artifact = get_store_resource(self.artifact)
+            else:
+                raise ValueError(
+                    "When passing artifact as a string, it must be a valid artifact store URI."
+                )
```

Comment:
I think we should consider moving this logic to a private method and only invoke it when the user explicitly calls `get_local_model_path` or `get_model_provider` in the future.